builtInFunctions/esdtModifyRoyalties.go

```diff
@@ -116,6 +125,16 @@ func (e *esdtModifyRoyalties) SetNewGasConfig(gasCost *vmcommon.GasCost) {
 	e.mutExecution.Unlock()
 }
 
+// SetBlockDataHandler is called when block data handler is set
+func (e *esdtModifyRoyalties) SetBlockDataHandler(blockDataHandler vmcommon.BlockDataHandler) error {
```

Comment:
so much duplicated code, this could have been part of a common component

Reply:
Refactored